src/main/java/org/sunbird/workflow/service/impl/UserProfileWfServiceImpl.java

```diff
@@ -64,11 +71,118 @@ public void updateUserProfile(WfRequest wfRequest) {
 	}
 
 	private void updateProfile(WfRequest wfRequest) {
-		StringBuilder builder = new StringBuilder();
-		String endPoint = configuration.getHubProfileUpdateEndPoint().replace(Constants.USER_ID_VALUE,
-				wfRequest.getApplicationId());
-		builder.append(configuration.getHubServiceHost()).append(endPoint);
-		requestServiceImpl.fetchResult(builder, wfRequest.getUpdateFieldValues(), Map.class);
+		try {
+			Map<String, Object> readData = (Map<String, Object>) userProfileRead(wfRequest.getApplicationId());
+			if (!readData.get(Constants.RESPONSE_CODE).equals(Constants.OK)) {
+				logger.error("user not found", ((Map<String, Object>) readData.get(Constants.PARAMS)).get(Constants.ERROR_MESSAGE));
+				wfRequest.setState(Constants.FAILED);
+				wfRequest.setAction(Constants.FAILED);
+				workflowService.workflowTransition(Constants.ROOT_ORG, Constants.ORG, wfRequest, wfRequest.getApplicationId());
+				return;
+			}
+			String existingDeptName = (String) ((Map<String, Object>) ((Map<String, Object>) ((Map<String, Object>) readData.get(Constants.RESULT)).get(Constants.RESPONSE)).get(Constants.ROOT_ORG_CONSTANT)).get(Constants.ORG_NAME);
+			String updatedDeptName = wfRequest.getDeptName();
+			if (!existingDeptName.equals(updatedDeptName)) {
+				Map<String, Object> Response = (Map<String, Object>) migrateUser(wfRequest);
+				if (!Response.get(Constants.RESPONSE_CODE).equals(Constants.OK)) {
+					logger.error("Migrate user failed", ((Map<String, Object>) Response.get(Constants.PARAMS)).get(Constants.ERROR_MESSAGE));
+					wfRequest.setState(Constants.FAILED);
+					wfRequest.setAction(Constants.FAILED);
+					workflowService.workflowTransition(Constants.ROOT_ORG, Constants.ORG, wfRequest, wfRequest.getApplicationId());
```

Comment:
these three lines are repeated many times... just create new private method and call it